photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala

```diff
@@ -178,12 +178,12 @@ class Driver(val params: Params, val sparkContext: SparkContext, val logger: Pho
     logger.info(s"Number of scored items: $numScoredItems (s)\n")
     val scoredItemsToBeSaved =
       if (numOutputFilesForScores > 0 && numOutputFilesForScores != scoredItems.partitions.length) {
-        scoredItems.coalesce(numOutputFilesForScores)
+        scoredItems.repartition(numOutputFilesForScores)
```

Comment:
Currently I don't see there would be a case like that for neither training nor scoring driver. And as you may already see from above comments, choosing ```repartition``` here for scoring driver is just because of that it gives us exactly the number of partitions specified by the user and thus is easier to test and understand by the user, and at the same time without introducing too much overhead compared to ```coalesce``` (which doesn't always give us the specified number of partitions even if the user wants to decrease the number of output files).